schema/sqlite/v3/visibility/schema.sql

```diff
@@ -1,24 +1,245 @@
 CREATE TABLE executions_visibility (
-	namespace_id CHAR(64) NOT NULL,
-	run_id CHAR(64) NOT NULL,
-	start_time TIMESTAMP NOT NULL,
-	execution_time TIMESTAMP NOT NULL,
-	workflow_id VARCHAR(255) NOT NULL,
-	workflow_type_name VARCHAR(255) NOT NULL,
-	status INT NOT NULL,  -- enum WorkflowExecutionStatus {RUNNING, COMPLETED, FAILED, CANCELED, TERMINATED, CONTINUED_AS_NEW, TIMED_OUT}
-	close_time TIMESTAMP NULL,
-	history_length BIGINT,
-	memo BLOB,
-	encoding VARCHAR(64) NOT NULL,
-	task_queue VARCHAR(255) DEFAULT '' NOT NULL,
-
-	PRIMARY KEY (namespace_id, run_id)
+  namespace_id        CHAR(64)      NOT NULL,
+  run_id              CHAR(64)      NOT NULL,
+  start_time          TIMESTAMP     NOT NULL,
+  execution_time      TIMESTAMP     NOT NULL,
+  workflow_id         VARCHAR(255)  NOT NULL,
+  workflow_type_name  VARCHAR(255)  NOT NULL,
+  status              INT           NOT NULL,  -- enum WorkflowExecutionStatus {RUNNING, COMPLETED, FAILED, CANCELED, TERMINATED, CONTINUED_AS_NEW, TIMED_OUT}
+  close_time          TIMESTAMP     NULL,
+  history_length      BIGINT        NULL,
+  memo                BLOB          NULL,
+  encoding            VARCHAR(64)   NOT NULL,
+  task_queue          VARCHAR(255)  NOT NULL DEFAULT '',
+  search_attributes   TEXT          NULL,
+
+  -- Predefined search attributes
+  TemporalChangeVersion         TEXT          GENERATED ALWAYS AS (JSON_EXTRACT(search_attributes, "$.TemporalChangeVersion")) STORED,
+  BinaryChecksums               TEXT          GENERATED ALWAYS AS (JSON_EXTRACT(search_attributes, "$.BinaryChecksums"))       STORED,
+  BatcherUser                   VARCHAR(255)  GENERATED ALWAYS AS (JSON_EXTRACT(search_attributes, "$.BatcherUser")),
+  TemporalScheduledStartTime    TIMESTAMP     GENERATED ALWAYS AS (STRFTIME('%Y-%m-%d %H:%M%:%f+00:00', JSON_EXTRACT(search_attributes, "$.TemporalScheduledStartTime"))),
+  TemporalScheduledById         VARCHAR(255)  GENERATED ALWAYS AS (JSON_EXTRACT(search_attributes, "$.TemporalScheduledById")),
+  TemporalSchedulePaused        BOOLEAN       GENERATED ALWAYS AS (JSON_EXTRACT(search_attributes, "$.TemporalSchedulePaused")),
+  TemporalNamespaceDivision     VARCHAR(255)  GENERATED ALWAYS AS (JSON_EXTRACT(search_attributes, "$.TemporalNamespaceDivision")),
```

Comment:
Is it going to be easy to add new generated fields here, e.g. if we add a new feature like schedules that uses a handful of new predefined search attributes? Will we need to add v4 of the schema or just write a small migration?

Reply:
The `v3` means sqlite3, not the schema version. We don't track schema version for sqlite, only for MySQL and PostgreSQL, in which case, you'll need to write scripts to add columns, etc.